test/e2e/custom_wait_rules_test.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package e2e
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestCustomWaitRules(t *testing.T) {
+	env := BuildEnv(t)
+	logger := Logger{}
+	kapp := Kapp{t, env.Namespace, env.KappBinaryPath, logger}
+	kubectl := Kubectl{t, env.Namespace, logger}
+
+	config := `
+apiVersion: kapp.k14s.io/v1alpha1
+kind: Config
+
+waitRules:
+  - ytt:
+      funcContractV1:
+        resource.star: |
+          def is_done(resource):
```

Comment:
is returning nothing the same (effectively) as returning `{}`? meaning done=false, successful=false, message='' ?

Reply:
Hi @jdef 
Yes, Here returning `{}` is equivalent to returning `{"done": False, "successful": False, "message": ""}` in such case it will keep on waiting until our default wait timeout of 15 minutes is not reached.